tools/textextractor/TextExtractor/Program.cs

```diff
@@ -0,0 +1,254 @@
+﻿namespace TextExtractor
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    using Windows.Globalization;
+    using Windows.Graphics.Imaging;
+    using Windows.Media.Ocr;
+
+    /// <summary>
+    /// Extracts share count from images using Windows 10 (https://blogs.windows.com/windowsdeveloper/2016/02/08/optical-character-recognition-ocr-for-windows-10/)
+    /// 
+    /// There are two options when running the utility
+    ///     TextExtractor.exe {directory} {outputfile}
+    ///         This syntax will enumerate through all images in a particular directory, analyze the images, and write results for each successful image to a
+    ///         comma-separated file (csv) with the format {fileName},{share count}
+    ///     TextExtractor.exe {file name}
+    ///         This syntax will allow processing of a single file. Output is written to the console. In the case of an error, the exit code will be non-zero and
+    ///         the reason for the failure will be written into the stderr stream.
+    /// </summary>
+    public static class Program
+    {
+        /// <summary>
+        /// The meat of the processing.
+        ///    1) loads an image
+        ///    2) converts it to grayscale (not sure if this is necessary)
+        ///    3) extracts text using windows UWP OCR
+        ///    4) enumerates line by line and parses floats
+        ///    5) does rudimentary logic to determine whether the value is reasonable
+        /// </summary>
+        /// <param name="fileInfo">File to process</param>
+        /// <returns>A tuple with the file name that was processed and the result.</returns>
+        private static async Task<Tuple<string, float>> ProcessSingleImageAsync(FileInfo fileInfo)
```

Comment:
You can return a named tuple which helps with readability. 
```csharp
private static async Task<(string fileName, float amount)> ProcessSingleImageAsync(...
```

then later
```csharp
return (fileInfo.Name, candidates[0])
```
and
```csharp
foreach (var (fileName, amount) in tasks.Where(t => t.IsCompletedSuccessfully).Select(t => t.Result))
{
    streamWriter.WriteLine($"{fileName},{amount}");
}
```